src/test/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigatorTest.java

```diff
@@ -115,11 +125,38 @@ public void fetchReposFromTeamSlug() throws Exception {
         List<SCMTrait<? extends SCMTrait<?>>> traits = new ArrayList<>(navigator.getTraits());
         traits.add(new TeamSlugTrait("justice-league"));
         navigator.setTraits(traits);
-        navigator.visitSources(SCMSourceObserver.filter(observer, "Hello-World", "github-branch-source-plugin"));
+        navigator.visitSources(SCMSourceObserver.filter(observer, "Hello-World", "github-branch-source-plugin", "unknown", "basic", "yolo", "yolo-archived"));
+
+        assertThat(projectNames, containsInAnyOrder("Hello-World", "github-branch-source-plugin", "basic", "yolo-archived"));
+    }
+
+    @Test
+    public void fetchOneRepoWithTeamSlug_InTeam() throws Exception {
+        final Set<String> projectNames = new HashSet<>();
+        final SCMSourceObserver observer = getObserver(projectNames);
+
+        List<SCMTrait<? extends SCMTrait<?>>> traits = new ArrayList<>(navigator.getTraits());
+        traits.add(new TeamSlugTrait("justice-league"));
+        navigator.setTraits(traits);
+        navigator.visitSources(SCMSourceObserver.filter(observer,   "yolo-archived"));
 
-        assertThat(projectNames, containsInAnyOrder("Hello-World", "github-branch-source-plugin"));
+        assertThat(projectNames, containsInAnyOrder( "yolo-archived"));
     }
 
+    @Test
+    public void fetchOneRepoWithTeamSlug_NotInTeam() throws Exception {
+        final Set<String> projectNames = new HashSet<>();
+        final SCMSourceObserver observer = getObserver(projectNames);
+
+        List<SCMTrait<? extends SCMTrait<?>>> traits = new ArrayList<>(navigator.getTraits());
+        traits.add(new TeamSlugTrait("justice-league"));
+        navigator.setTraits(traits);
+        navigator.visitSources(SCMSourceObserver.filter(observer, "yolo"));
```

Comment:
The actual test for this feature. When you only have a single repo there is a different code path for filtering.